tuxemon/core/components/event/actions/combat.py

```diff
@@ -213,6 +213,9 @@ def random_encounter(self, game, action):
         if not self.check_battle_legal(player1):
             return False
 
+        #stop player motion
+        player1.moving = False
```

Comment:
I think the core issue should be addressed....that it seems the order between player movement, keypresses, and event handling could be rearranged to prevent anomalies like this.  I'm sure there are other quirks related to it, but they are not obvious like this.